### PR TITLE
Dispatcher behaviour in Flatpak sandbox

### PR DESCRIPTION
**Pull request by ykgmfq**
This series is a first attempt to adjust the handling of dispatchers when inside a sandbox. Related to https://github.com/qutebrowser/qutebrowser/issues/4879

### CODE REVIEW

File: qutebrowser/utils/version.py [hunk 1]
```diff
@@ -123,6 +123,15 @@ def distribution():
                             id=dist_id)
 
 
+def sandboxed():
+    """Check whether the environment is a sandbox, i.e. has restricted access to the host system.
+
+    Return:
+        True if the environment is a sandbox.
+    """
+    return distribution().parsed == Distribution.kde
```
Comment (The-Compiler) [inline qutebrowser/utils/version.py]: `distribution()` can be `None` (e.g. on non-Linux systems)

File: qutebrowser/utils/version.py [hunk 2]
```diff
@@ -123,6 +123,15 @@ def distribution():
                             id=dist_id)
 
 
+def sandboxed():
```
Comment (The-Compiler) [inline qutebrowser/utils/version.py]: I'd prefer `is_sandboxed()`

File: qutebrowser/utils/version.py [hunk 3]
```diff
@@ -123,6 +123,15 @@ def distribution():
...
+    """Check whether the environment is a sandbox, i.e. has restricted access to the host system.
+
+    Return:
+        True if the environment is a sandbox.
+    """
```
Comment (The-Compiler) [inline qutebrowser/utils/version.py]: Can probably drop that docstring, as the function is simple enough to not need any explanation.

**Comment by ykgmfq** [comment]
@The-Compiler done, I think

**Comment by The-Compiler** [comment]
I just noticed there are tests for `open_file` in `tests/unit/utils/test_utils.py` as well - can you please add new testcases to cover the sandboxed code?

Also, there are some [flake8](https://travis-ci.org/qutebrowser/qutebrowser/jobs/556776452)/[pylint](https://travis-ci.org/qutebrowser/qutebrowser/jobs/556776451) things to fix.

Let me know if you need help somewhere!

**Comment by The-Compiler** [comment]
I merged and finished this now, with the following follow-up commits:

- 22503c70a Fix lint
- 3a4f08c0c Complete tests for utils.open_file with sandbox
- f6e0d1af7 Rename Distribution.kde to .kde_flatpak

Thanks!

**Pull request merged**